core/src/main/scala/org/bitcoins/core/p2p/ServiceIdentifier.scala

```diff
@@ -173,6 +173,16 @@ object ServiceIdentifier extends Factory[ServiceIdentifier] {
   def fromBytes(bytes: ByteVector): ServiceIdentifier =
     RawServiceIdentifierSerializer.read(bytes)
 
+  def fromString(string: String): ServiceIdentifier = string match {
```

Comment:
let's add a catch-all case `_: String` and throw an `IllegalArgumentException` with a more descriptive error message than `unmatched` which is what would happen as is